src/fmu/dataio/providers/_objectdata.py

```diff
@@ -0,0 +1,676 @@
+    content: depth
+
+    # tagname is flexible. The tag is intended primarily for providing uniqueness.
+    # The tagname will also be part of the outgoing file name on disk.
+    tagname: ds_extract_geogrid
+
+    # no content-specific attribute for "depth" but can come in the future
+
+    properties: # what the values actually show. List, only one for IRAP Binary
+                # surfaces. Multiple for 3d grid or multi-parameter surfaces.
+                # First is geometry.
+        - name: PropertyName
+          attribute: owc
+          is_discrete: false # to be used for discrete values in surfaces.
+          calculation: null # max/min/rms/var/maxpos/sum/etc
+
+    format: irap_binary
+    layout: regular # / cornerpoint / structured / etc
+    unit: m
+    vertical_domain: depth # / time / null
+    depth_reference: msl # / seabed / etc # mandatory when vertical_domain is depth?
+    grid_model: # Making this an object to allow for expanding in the future
+        name: MyGrid # important for data identification, also for other data types
+    spec: # class/layout dependent, optional? Can spec be expanded to work for all
+          # data types?
+        ncol: 281
+        nrow: 441
+        ...
+    bbox:
+        xmin: 456012.5003497944
+        xmax: 467540.52762886323
+        ...
+
+    # --- NB two variants of time, here old:
+    time:
+        - value: 2029-10-28T11:21:12
+          label: "some label"
+        - value: 2020-10-28T14:28:02
+          label: "some other label"
+
+    # --- Here new:
+    t0:
+        value: 2020-10-28T14:28:02
+        label: "some other label"
+    t1:
+        value: 2029-10-28T11:21:12
+        label: "some label"
+
+    is_prediction: true # For separating pure QC output from actual predictions
+    is_observation: true # Used for 4D data currently but also valid for other data?
+    description:
+        - Depth surfaces extracted from the structural model
+
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Final
+
+import numpy as np
+import pandas as pd
+import xtgeo
+
+from fmu.dataio._definitions import STANDARD_TABLE_INDEX_COLUMNS, ValidFormats
+from fmu.dataio._logging import null_logger
+from fmu.dataio.datastructure.meta import meta, specification
+
+from ._objectdata_base import (
+    DerivedObjectDescriptor,
+    ObjectDataProvider,
+    SpecificationAndBoundingBox,
+)
+
+if TYPE_CHECKING:
+    from fmu.dataio.dataio import ExportData
+    from fmu.dataio.types import Inferrable
+
+logger: Final = null_logger(__name__)
+
+
+def objectdata_provider_factory(
+    obj: Inferrable, dataio: ExportData, meta_existing: dict | None = None
+) -> ObjectDataProvider:
+    """
+    obj: types.Inferrable
+    dataio: dataio.ExportData
+    meta_existing: dict = field(default_factory=dict)
+    """
+    if meta_existing:
+        return ExistingDataProvider(obj=obj, dataio=dataio, meta_existing=meta_existing)
```

Comment:
Since we short circuit on existing metadata here it doesn't make sense to pass it in all other cases below. This will be changed in a forthcoming PR.